### DISCUSSION TITLE
How to link a pdf which opens a pdf-viewer in GitHub pages

### DISCUSSION THREAD

**Discussion by fabitmart** [community/community#21720] [Other Feature Feedback, Questions, & Ideas]
<p>I am very new to GitHub and to web developing. I am creating a static personal page on GitHub pages. I have a set of slides and I managed to link to them using a picture. However, I am stuck with the solution below, which opens my directory and previews the file but does not open the slides nicely in a pdf-viewer.</p>
<pre><code>&lt;a href="https://github.com/fabfabit/MyStuff_public/blob/master/files/ProjMarr_slides.pdf" class="image fit" &gt;&lt;img src="images/marr_pic.jpg" alt=""&gt;&lt;/a&gt;
</code></pre>
<p>I am too noob to understand the solutions below:</p>
<p><a href="https://stackoverflow.com/questions/30745981/opening-pdf-in-a-browser-with-github-pages" rel="nofollow noopener">Opening PDF in a browser with Github Pages</a></p>
<p><a href="https://webapps.stackexchange.com/questions/48061/can-i-trick-github-into-displaying-the-pdf-in-the-browser-instead-of-downloading" rel="nofollow noopener">https://webapps.stackexchange.com/questions/48061/can-i-trick-github-into-displaying-the-pdf-in-the-browser-instead-of-downloading</a></p>
<p>The only suspicion that I have is that this might be due to the fact that the website has not been published yet, but I am just speculating. Could anyone explain to me how to fix this?</p>
<p>Crossposting from <a href="https://stackoverflow.com/questions/56359071/how-to-link-a-pdf-which-opens-a-pdf-viewer-in-github-pages" rel="nofollow noopener">here</a></p>

**Comment by Gisgar3** [comment]
<p>From the HTML code that you provided, it looks like you’re trying to grab a PDF from a separate GitHub repository that is non-relating to your <strong>GitHub Pages </strong> one. To fix this issue, it is recommended that you upload the PDF to your main  <strong>GitHub Pages</strong> repository and then referencing it in your HTML code. For example:</p>
<pre><code>&lt;a href="/pdfs/ProjMarr_slides.pdf" class="image fit"&gt;&lt;img src="images/marr_pic.jpg" alt=""&gt;&lt;/a&gt;
</code></pre>
<p>^ this is also assuming you put the PDF file in its own respective <em>pdfs</em> folder. If you still have a problem getting it to open correctly, try adding this to your <em>&lt;a&gt;</em> tag:</p>
<pre><code>type="application/pdf"
</code></pre>
<p>Hope this works out well!</p>

**Comment by fabitmart** [comment, marked as answer]
<p>Thank you so much, you have been most helpful and kind.</p>
<p>For those who might be reading this because of a similar issue, I want to point out two minor things:</p>
<ol>
<li>The depository was already the one pertaining to my GitHub Pages. The issue was that by calling it as above, I was (I think) effectively treating it as an external depository;</li>
<li>Once I used the correct method for calling the file, it worked. The only thing I had  to change (relative to your suggestion) was that I had to delete the firts “/” you put in your path.</li>
</ol>
<p>The working solution was:</p>
<pre><code>&lt;a href="pdfs/ProjMarr_slides.pdf" class="image fit"&gt;&lt;img src="images/marr_pic.jpg" alt=""&gt;&lt;/a&gt;
</code></pre>
<p>instead of</p>
<pre><code>&lt;a href="/pdfs/ProjMarr_slides.pdf" class="image fit"&gt;&lt;img src="images/marr_pic.jpg" alt=""&gt;&lt;/a&gt;
</code></pre>
<p>Thank you so much for your help.</p>
<p>Fabio</p>

**Reply by phuocnguyen2201** [reply]
Still working after 5 years

**Reply by annimukherjee** [reply]
Yep!
Love GitHub ❤️

**Comment by Xelliese** [comment]
pening nicely, make sure your GitHub Pages site is published. If it's already published, try adding target="_blank" to your link to open the PDF in a new tab. If that doesn't work, consider using a library like PDF.js to embed the PDF directly into your page.

**Comment by tomcomtang** [comment]
I know a very simple method: [https://pages.edgeone.ai/use-cases/pdf-to-url](https://pages.edgeone.ai/use-cases/pdf-to-url) You can directly drag and drop your PDF file to upload it, and it will instantly generate an accessible HTML link. The only step required is to sign up quickly using Google. Hope this helps you.

**Reply by mrtysn** [reply]
For me, the whole purpose of hosting on GitHub pages is to keep it self-contained and auto-deployable. Externally hosting a PDF is not the solution I'd be opting for here.
